Solutions/Marain.Operations.Api.Specs/Bindings/OperationsContainerBindings.cs

```diff
@@ -0,0 +1,90 @@
+﻿// <copyright file="OperationsContainerBindings.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Operations.Api.Specs.Bindings
+{
+    using System;
+    using Corvus.Configuration;
+    using Corvus.Identity.ManagedServiceIdentity.ClientAuthentication;
+    using Corvus.Testing.SpecFlow;
+    using Marain.Operations.Client.OperationsControl;
+    using Marain.Tenancy.Client;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using TechTalk.SpecFlow;
+
+    [Binding]
+    public static class OperationsContainerBindings
+    {
+        [BeforeFeature(Order = ContainerBeforeFeatureOrder.PopulateServiceCollection)]
+        public static void PerFeatureContainerSetup(FeatureContext featureContext)
+        {
+            ContainerBindings.ConfigureServices(
+                featureContext,
+                services =>
+                {
+                    var configBuilder = new ConfigurationBuilder();
+                    configBuilder.AddConfigurationForTest("appsettings.json");
+                    IConfigurationRoot config = configBuilder.Build();
+                    services.AddSingleton<IConfiguration>(config);
+
+                    services.AddLogging(x =>
+                    {
+                        x.AddConsole();
+                    });
+
+                    services.AddJsonNetSerializerSettingsProvider();
+                    services.AddJsonNetPropertyBag();
+                    services.AddJsonNetCultureInfoConverter();
+                    services.AddSingleton<JsonConverter>(new StringEnumConverter(true));
+
+                    // Tenancy service client.
+                    services.AddSingleton(sp =>
+                    {
+                        TenancyClientOptions tenancyConfiguration = sp.GetRequiredService<IConfiguration>().GetSection("TenancyClient").Get<TenancyClientOptions>();
+
+                        if (tenancyConfiguration?.TenancyServiceBaseUri == default)
```

Comment:
This is a slightly verbose way of comparing with `null`, right? If so: 1) why not compare with just `null`, or 2) why not `string.IsNullOrWhitespace` (or whatever it's called)? Or is "present but empty" a valid setting?

Reply:
No, "present but empty" isn't valid. `tenancyConfiguration.TenancyServiceBaseUri` is a `Uri`, not a `string`, so `string.IsNullOrWhitespace` doesn't work, but I've just verified that if it's an empty string in config, the resultant value in the property will be null, so have updated to use `is null` instead.